retriever/lib/datasets.py

```diff
@@ -35,7 +35,8 @@ def datasets(keywords=None, licenses=None):
     result_scripts_offline = sorted(list(result_scripts_offline), key=lambda s: s.name.lower())
     result_retriever_scripts_online = get_dataset_names_upstream(keywords, licenses, repo=RETRIEVER_REPOSITORY)
     result_scripts_online = get_dataset_names_upstream(keywords, licenses)
-    return dict({'online': sorted(result_retriever_scripts_online + result_scripts_online), 'offline': result_scripts_offline})
+    return dict({'online': sorted(result_retriever_scripts_online + result_scripts_online), \
+        'offline': result_scripts_offline})
```

Comment:
Also in `def dataset_names()`
` scripts_name = dict({'online': [], 'offline': []})` can be written as
`scripts_name = {'online': [], 'offline': []}`

Reply:
@harshitbansal05, I have made some draft sample for you to reference.

```python
from retriever.lib.defaults import RETRIEVER_REPOSITORY
from retriever.lib.scripts import SCRIPT_LIST, get_script, get_dataset_names_upstream


def datasets(keywords=None, licenses=None):
    """Search all datasets by keywords and licenses."""
    script_list = SCRIPT_LIST()
    if not keywords and not licenses:
        # comment
        offline_scripts = sorted(script_list, key=lambda s: s.name.lower())
        # comment
        on_demand_scripts = get_dataset_names_upstream(repo=RETRIEVER_REPOSITORY)
        # comment
        upstream_scripts = get_dataset_names_upstream()
        # comment
        native_scripts = sorted(on_demand_scripts + upstream_scripts)
        
        return {'online': native_scripts, 'offline': offline_scripts}

    result_scripts_offline = set()
    if licenses:
        licenses = [l.lower() for l in licenses]
    for script in script_list:
        if script.name:
            if licenses:
                script_license = [licence_map['name'].lower()
                                  for licence_map in script.licenses
                                  if licence_map['name']]
                if script_license and set(script_license).intersection(set(licenses)):
                    result_scripts_offline.add(script)
                    continue
            if keywords:
                script_keywords = script.title + ' ' + script.name
                if script.keywords:
                    script_keywords = script_keywords + ' ' + '-'.join(script.keywords)
                script_keywords = script_keywords.lower()
                for k in keywords:
                    if script_keywords.find(k.lower()) != -1:
                        result_scripts_offline.add(script)
                        break
    # comment about the values here and use short names. 
    result_scripts_offline = sorted(list(result_scripts_offline), key=lambda s: s.name.lower())
    # comment 
    result_retriever_scripts_online = get_dataset_names_upstream(keywords, licenses, repo=RETRIEVER_REPOSITORY)
    # comment 
    result_scripts_online = get_dataset_names_upstream(keywords, licenses)

    all_online = sorted(result_retriever_scripts_online + result_scripts_online)
    result = {'online': all_online, 'offline': result_scripts_offline}
    return result


def dataset_names():
    """Return list of all available dataset names."""
    all_scripts = datasets()
    scripts_name = {'online': [], 'offline': []}
    for offline_script in all_scripts['offline']:
        scripts_name['offline'].append(offline_script.name)
    for online_script in all_scripts['online']:
        scripts_name['online'].append(online_script)
    return scripts_name


def license(dataset):
    """Get the license for a dataset."""
    return get_script(dataset).licenses[0]['name']


def dataset_licenses():
    """Return set with all available licenses."""
    license_values = [str(script.licenses[0]['name']).lower()
                      for script in SCRIPT_LIST()]
    return set(license_values)

```